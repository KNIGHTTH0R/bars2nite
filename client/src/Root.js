import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import reduxThunk from 'redux-thunk';
import reducers from './store/reducers';

const composedEnhancers =
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default ({ children, initialState = {} }) => {
  const store = createStore(
    reducers,
    initialState,
    composedEnhancers(applyMiddleware(reduxThunk))
  );

  return <Provider store={store}>{children}</Provider>;
};
