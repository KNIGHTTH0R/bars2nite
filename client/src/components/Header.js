import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../store/actions';
import SideNav from './SideNav';

export class Header extends Component {
  state = {
    searchInput: '',
    sideNavShow: false
  };

  inputChangeHandler = e => {
    this.setState({ searchInput: e.target.value });
  };

  keyPressHandler = e => {
    if (e.keyCode === 13) {
      e.preventDefault();
      this.onSearchBars();
    }
  };

  toggleSideNav = () => {
    this.setState(prevState => {
      return { sideNavShow: !prevState.sideNavShow };
    });
  };

  onSearchBars = () => {
    if (this.state.searchInput !== '') {
      this.props.onSearchBars(this.state.searchInput);
      this.props.history.push('/search');
      this.setState({ searchInput: '' });
    }
  };

  render() {
    return (
      <div className="Header">
        <nav className="navigation">
          <span className="navigation__header">
            <Link to="/">
              <i className="fa fa-glass glass-icon--1" aria-hidden="true" />
              <i className="fa fa-glass glass-icon--2" aria-hidden="true" />
            </Link>
          </span>
          <form action="#" className="search">
            <input
              type="text"
              className="search__input"
              placeholder="Your city or zip code"
              value={this.state.searchInput}
              onChange={this.inputChangeHandler}
              onKeyDown={this.keyPressHandler}
            />
            <i
              className="fa fa-search search__icon"
              aria-hidden="true"
              onClick={this.onSearchBars}
            />
          </form>

          <ul className="navigation__list">
            <li className="sidenav-icon">
              <i
                onClick={this.toggleSideNav}
                className="fa fa-bars"
                aria-hidden="true"
              />
            </li>

            <li
              className="SideNav"
              style={{
                marginRight: this.state.sideNavShow ? '0' : '-150px',
                opacity: this.state.sideNavShow ? 1 : 0
              }}
            >
              <SideNav />
            </li>
          </ul>
        </nav>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    auth: state.auth
  };
}

export default connect(
  mapStateToProps,
  actions
)(withRouter(Header));
