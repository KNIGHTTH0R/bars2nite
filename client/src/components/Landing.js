import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../store/actions';

import mp4Video from '../assets/video.mp4';
import webmVideo from '../assets/video.webm';

export class Landing extends Component {
  componentDidMount = () => {
    this.props.onLoadReservedBars();
  };

  render() {
    return (
      <div className="Landing">
        <div className="bg-video">
          <video className="bg-video__content" autoPlay loop>
            <source src={mp4Video} type="video/mp4" />
            <source src={webmVideo} type="video/webm" /> Your browser is not
            supported!
          </video>
        </div>

        <div className="header-box">
          <h2 className="header-box__primary">Bars2Nite</h2>
          <p className="header-box__secondary">rspv now</p>
        </div>
      </div>
    );
  }
}

export default connect(
  null,
  actions
)(Landing);
