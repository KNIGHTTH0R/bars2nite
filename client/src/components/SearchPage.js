import React, { Component } from 'react';
import { connect } from 'react-redux';
import SearchItem from '../helpers/SearchItem';

export class SearchPage extends Component {
  render() {
    let searchedBars;
    if (this.props.bars) {
      searchedBars = this.props.bars.map(bar => {
        return (
          <SearchItem
            key={bar.id}
            name={bar.name}
            img={bar.image_url}
            price={bar.price}
            rating={bar.rating}
          />
        );
      });
    }

    // console.log(this.props.bars);
    return <div className="SearchPage">{searchedBars}</div>;
  }
}

const mapStateToProps = state => {
  return {
    bars: state.bars.searchedBars
  };
};

export default connect(mapStateToProps)(SearchPage);
