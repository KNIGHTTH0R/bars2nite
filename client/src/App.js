import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from './store/actions';
import Header from './components/Header';
import Landing from './components/Landing';
import SearchPage from './components/SearchPage';
import Events from './components/Events';
import AllEvents from './components/AllEvents';

// testing purpose
import axios from 'axios';
window.axios = axios;

class App extends Component {
  componentDidMount = () => {
    this.props.fetchUser();
  };

  render() {
    return (
      <div className="App">
        <BrowserRouter>
          <div className="container">
            <Header />
            <Route path="/" exact component={Landing} />
            <Route path="/search" exact component={SearchPage} />
            <Route path="/events" exact component={Events} />
            <Route path="/all-events" exact component={AllEvents} />
          </div>
        </BrowserRouter>
      </div>
    );
  }
}

export default connect(
  null,
  actions
)(App);
