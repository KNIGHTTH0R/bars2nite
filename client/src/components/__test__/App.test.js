import React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../../App';
import Root from '../../Root';
import Header from '../Header';
import Landing from '../Landing';
import SearchPage from '../SearchPage';

// import { fetchUser } from '../../store/actions';
// import * as actionTypes from '../../store/actions/actionTypes';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <App />
    </Root>
  );
});

describe('App', () => {
  const app = shallow(
    <Root>
      <App />
    </Root>
  );

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });

  // it('contains a Landing component', () => {
  //   expect(wrapped.find(Landing).length).toEqual(1);
  //   wrapped.unmount();
  // });

  // it('contains a Header component', () => {
  //   expect(wrapped.find(Header).length).toEqual(1);
  //   wrapped.unmount();
  // });

  // it('contains a SearchPage component', () => {
  //   console.log(wrapped);
  //   expect(app.find('Connect(SearchPage)').exists()).toBe(true);
  //   wrapped.unmount();
  // });
});
