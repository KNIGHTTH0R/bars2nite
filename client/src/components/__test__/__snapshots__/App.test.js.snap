// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter>
        <div
                className="container"
        >
                <Connect(withRouter(Header)) />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/search"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/events"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/all-events"
                />
        </div>
</BrowserRouter>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="container"
>
          <Connect(withRouter(Header)) />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/search"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/events"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/all-events"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(withRouter(Header)) />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/search"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/events"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/all-events"
/>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/events",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/all-events",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <div
                    className="container"
          >
                    <Connect(withRouter(Header)) />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/search"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/events"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/all-events"
                    />
          </div>
</BrowserRouter>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="container"
>
            <Connect(withRouter(Header)) />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/search"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/events"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/all-events"
            />
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(withRouter(Header)) />,
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/search"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/events"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/all-events"
/>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/events",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/all-events",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
